echo-pubsub-google/src/main/java/com/netflix/spinnaker/echo/config/GooglePubsubProperties.java

```diff
@@ -79,6 +83,29 @@ public InputStream readTemplatePath() {
     }
   }
 
+  @Data
+  @NoArgsConstructor
+  public static class GooglePubsubPublisherConfig {
+
+    @NotEmpty
+    private String name;
+
+    @NotEmpty
+    private String project;
+
+    // Optional. Uses Application Default Credentials if not set.
```

Comment:
Are "application default credentials" the credentials that echo would get from clouddriver?

Reply:
[App default credentials](https://cloud.google.com/docs/authentication/production#providing_credentials_to_your_application) are a feature of GCP - client libraries can grab credentials from the environment if running in GCP